Update AngularFire v7 & Firebase v9

randallmeeker opened the pull request:
resolves #205

GrandSchtroumpf committed:
cache query & use share delay

GrandSchtroumpf committed:
cache query & use share delay

GrandSchtroumpf committed:
keep memory of last value seen

randallmeeker committed:
Merge remote-tracking branch 'origin/v6.0.4' into firebase-v9

randallmeeker committed:
Update prettier

randallmeeker committed:
Update prettier

randallmeeker committed:
add rxfire

randallmeeker committed:
add rxfire

randallmeeker committed:
update import paths

randallmeeker committed:
update import paths

randallmeeker commented:
Is there a prettier setup that should be used? I'm attempting to add rules that will limit all the white space changes to correspond with the formatting I see already present in the repo.

randallmeeker commented:
@GrandSchtroumpf sorry mean to tag earlier.

randallmeeker committed:
fix collection service typing

randallmeeker committed:
fix collection service typing

GrandSchtroumpf commented:
There is no prettier setup because I never managed to achieve something versatile enough for HTML, CSS & TS with prettier. I consider it ~impossible~ very difficult to enforce code understandability with the line width as main parameter. For example [this update](https://github.com/dappsnation/akita-ng-fire/pull/217/files#diff-daa072188c090d5579d7f0fa56a7f54788c484c63fb1a7486199cc5459afa857R69) or [this one](https://github.com/dappsnation/akita-ng-fire/pull/217/files#diff-b52c7395f4d0a58bf5962c0e5f6a4f66c3bad16286044cdadabd926dd48e6fbeR30) look less understandable for me than the previous ones.
I've no issue with adding prettier setup, but I would recommend to not mix PR purpose 🙂.

randallmeeker commented:
I get that, I was gonna suggest a pre-PR where I just run whatever the Prettier defaults are already in the repo across the entire thing, then this merge into this.

randallmeeker committed:
re-run prettier

randallmeeker committed:
re-run prettier

randallmeeker committed:
Merge remote-tracking branch 'origin/v6' into firebase-v9

randallmeeker committed:
prettier

randallmeeker committed:
prettier

randallmeeker committed:
Merge branch 'v6' into firebase-v9

randallmeeker committed:
update ivy settings, lock and deps

randallmeeker committed:
update ivy settings, lock and deps

randallmeeker commented:
@GrandSchtroumpf 

This builds and runs. However, the test suite does not seem to run, but this was the case before I started. Let me know if there are any suggestions there.

I did this off the 6.0.4 brach and as there were changes since v6 that needed updates.

Aside from getting the test suite up and running, I was going to try to upgrade one of my current projects with it, but it appears good to go.

GrandSchtroumpf commented on package.json:
```diff
@@ -31,7 +31,8 @@
     "akita-ng-fire": "^6.0.0",
-    "firebase": "^8.0.1",
+    "firebase": "^9.1.0",
+    "rxfire": "^6.0.2",
```
is rxfire a perrDependancy of @angular/fire ?

randallmeeker replied:
It is. `ng add @angular/fire` automatically add it as will `ng update @angular/fire`.

GrandSchtroumpf commented on tsconfig.base.json:
```diff
@@ -20,8 +20,5 @@
       "akita-ng-fire/*": ["dist/akita-ng-fire/*"]
     }
-  },
-  "angularCompilerOptions": {
```
Is @angular/fire dependent of Ivy ? As far as I remember, it's still consider experimental to build a lib with Ivy

randallmeeker replied:
ivy is forced enabled in angular 12, this line was telling us it was turning it off, but in reality its now always on. It was flashing up warnings that we were attempting to turn it off so I simply removed this.

GrandSchtroumpf commented:
@randallmeeker I've updated the lib to fix this issue #216. I think you'll need to rebase your branch on v6 to have the latest version.
Also I think this is a major update, therefore we should bump the version to 7.0.0.
What we can do for the testing the lib would be to publish this with a next tag. To achieve that, you can 
- bump the version of package.json to `7.0.0-beta.1`
- create a PR against the [branch v7](https://github.com/dappsnation/akita-ng-fire/tree/v7)
- once it's merged, I'll publish version with the tag `next`.
- from there you can update your projects and do the required checks.

randallmeeker committed:
Merge branch 'firebase-v9' of github.com:randallmeeker/akita-ng-fire into firebase-v9

randallmeeker committed:
bump v7 version

GrandSchtroumpf commented:
outch I think something went wrong with the rebase here. The commits are in double and you've got conflicts.
I would recommend a new branch fresh from `v7` and cherry pick your commits. I think it would be faster than trying to clean the history / conflicts.
